pkg/ddc/jindofsx/transform.go

```diff
@@ -344,44 +352,35 @@ func (e *JindoFSxEngine) transformMaster(runtime *datav1alpha1.JindoRuntime, met
 		for _, encryptOption := range mount.EncryptOptions {
 			key := encryptOption.Name
 			secretKeyRef := encryptOption.ValueFrom.SecretKeyRef
-			secret, err := kubeclient.GetSecret(e.Client, secretKeyRef.Name, e.namespace)
-			if err != nil {
-				e.Log.Info("can't get the secret")
-				break
-			}
-			value := secret.Data[secretKeyRef.Key]
-			if err != nil {
-				e.Log.Info("decode value failed")
-			}
-			if key == "fs.oss.accessKeyId" {
-				propertiesFileStore["jindofsx.oss.accessKeyId"] = string(value)
-			}
-			if key == "fs.oss.accessKeySecret" {
-				propertiesFileStore["jindofsx.oss.accessKeySecret"] = string(value)
-			}
-			if key == "fs.s3.accessKeyId" {
-				propertiesFileStore["jindofsx.s3.accessKeyId"] = string(value)
-			}
-			if key == "fs.s3.accessKeySecret" {
-				propertiesFileStore["jindofsx.s3.accessKeySecret"] = string(value)
-			}
-			if key == "fs.cos.accessKeyId" {
-				propertiesFileStore["jindofsx.cos.accessKeyId"] = string(value)
-			}
-			if key == "fs.cos.accessKeySecret" {
-				propertiesFileStore["jindofsx.cos.accessKeySecret"] = string(value)
-			}
-			if key == "fs.obs.accessKeyId" {
-				propertiesFileStore["jindofsx.obs.accessKeyId"] = string(value)
-			}
-			if key == "fs.obs.accessKeySecret" {
-				propertiesFileStore["jindofsx.obs.accessKeySecret"] = string(value)
+			if secretMountSupport {
+				value.Secret = secretKeyRef.Name
+				if key == "fs."+mountType+".accessKeyId" {
+					value.SecretKey = key
+					e.Log.Info("Get %s From %s!", key, secretKeyRef.Name)
+				}
+				if key == "fs."+mountType+".accessKeySecret" {
+					value.SecretValue = key
+					e.Log.Info("Get %s From %s!", key, secretKeyRef.Name)
+				}
+			} else {
+				secret, err := kubeclient.GetSecret(e.Client, secretKeyRef.Name, e.namespace)
+				if err != nil {
+					e.Log.Info("can't get the input secret from dataset", secretKeyRef.Name)
+					break
+				}
+				value := secret.Data[secretKeyRef.Key]
+				if key == "fs."+mountType+".accessKeyId" {
+					propertiesFileStore["jindofsx."+mountType+".accessKeyId"] = string(value)
```

Comment:
Why this key here is like `jindofsx.oss.accessKeyId` but in the code above it stores properties like `jindofsx.oss.bucket."+bucketName+".accessKeyId"`. Does Jindofsx support both?

Reply:
BTW, can we support multiple bucket with different ak/sk now?